interfaces/apparmor/backend.go

```diff
@@ -216,7 +217,11 @@ func (b *Backend) deriveContent(spec *Specification, snapInfo *snap.Info, opts i
 
 func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.ConfinementOptions, snippetForTag string, content map[string]*osutil.FileState) {
 	var policy string
-	if opts.Classic && !opts.JailMode {
+	// When partial AppArmor is detected, use the classic template for now. We could
+	// use devmode, but that could generate confusing log entries for users running
+	// snaps on systems with partial AppArmor support.
+	level, _ := aa.ProbeKernel().Evaluate()
```

Comment:
Agree with Michael: this interface is too complicated for how trivial this is. What is the expected result of probing for a kernel and what does it mean to _Evaluate_ it?

How about: _aa.SupportLevel() == aa.PartialSupport_?

The fact we need to compute and cache should remain inside the apparmor package itself. The _release_ package may help with some ideas on that regard.

Also, what would "partial" mean in this case, and why do we expect what is inside classic to work while what is in strict won't? It would be better to be more clear about what is supported on not, and if possible have that gradient explicit in the terminology.

As a side note, Michael is _extremely_ polite.. when he raises something on my code or ideas, I think three times before dismissing it.

Reply:
In fact, that's all the apparmor package does right now. This would be better as release.ApparmorSupport instead, I think.

Reply:
Please bare with me, the result of ProbeKernel will be stored in the apparmor backend. From there we will call Evaluate in one spot and access available features in several spots. I can bring this patch earlier. Partial means, as it is documented, that apparmor is enabled but some of the required features are missing. As discussed with jdstrand classic is used because it works on every release of apparmor and allows us to take the first step. The next step will be to switch to the actual strict template and teach specific interfaces about any extra deficiencies or incompatibilities that resulted from how apparmor enforcement works when a particular feature is present or absent. We already have the gradient you are talking about - we can ask, through `SupportsFeature` if a given feature is enabled or not.

Reply:
@zyga I understood this proposal from your earlier comments. My points above already take it into account.

Reply:
`release.ApparmorSupports("feature")` sounds nice, btw.